### PR TITLE
Add user model, dao, service and authentication service

### PR DESCRIPTION
**Pull request by OleksiiTromsa**

### CODE REVIEW

**Review by DmGrt — APPROVED** [review]
It looks like everything is fine

File: src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        byte[] salt = HashUtil.getSalt();
+        user.setPassword(HashUtil.hashPassword(user.getPassword(), salt));
+        user.setSalt(salt);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java]:
```suggestion
        user.setSalt(HashUtil.getSalt());
        user.setPassword(HashUtil.hashPassword(user.getPassword(), user.getSalt()));
```

File: src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            return user.get();
+        }
+        throw new AuthenticationException("Incorrect email or password");
+    }
+
+    @Override
+    public User register(String email, String password) throws AuthenticationException {
+        if (email.isEmpty() || password.isEmpty()) {
+            throw new AuthenticationException("Email or password is empty");
+        }
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java]: let's remove this check, we will validate input later on in controller + you throw AuthenticationException while registration - these are different procedures

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        }
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<User> query = session.createQuery(
+                    "FROM User WHERE email = :email", User.class)
+                    .setParameter("email", email);
+            return Optional.ofNullable(query.uniqueResult());
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java]:
```suggestion
            return query.uniqueResultOptional();
```

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(user);
+            transaction.commit();
+            return user;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert User entity", e);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java]: let's make message more informative and add info about user

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**